ui/src/layout/signin/RootComponent.js

```diff
@@ -27,22 +27,16 @@ const AnimationItem = (props) => {
   )
 }
 
-@withTranslation()
-class TiDBSignInForm extends React.PureComponent {
-  state = {
-    loading: false,
-    signInError: null,
-  }
+function TiDBSignInForm({ t, registry }) {
```

Comment:
Maybe you can directly use the `t` by `const { t } = useTranslation()` inside the component, so you don't need to pass it via props by `withTranslation()(App)`, you can have a try.

Reply:
Done.